cmdk/src/index.tsx

```diff
@@ -574,10 +590,23 @@ const Command = React.forwardRef<HTMLDivElement, CommandProps>((props, forwarded
       >
         {label}
       </label>
-      <StoreContext.Provider value={store}>
-        <CommandContext.Provider value={context}>{children}</CommandContext.Provider>
-      </StoreContext.Provider>
-    </div>
+      {props.asChild ? (
+        <Slottable>
+          {React.isValidElement(children) &&
+            React.cloneElement(
+              children,
+              undefined,
+              <StoreContext.Provider value={store}>
+                <CommandContext.Provider value={context}>{children.props.children}</CommandContext.Provider>
+              </StoreContext.Provider>,
+            )}
+        </Slottable>
+      ) : (
+        <StoreContext.Provider value={store}>
+          <CommandContext.Provider value={context}>{children}</CommandContext.Provider>
+        </StoreContext.Provider>
+      )}
```

Comment:
I had to do this in some components because `Slot` does not support when children are nested. For example:

If we have this component:
```tsx
const Button = () => (
  <Slot>
    <div>
      {children}
    </div>
  </Slot>
)
```

and using like
```tsx
<Button asChild>
  <a>
   Button
  </a>
</Button>
```

will result in
```tsx
<div>
  <a>
   Button
  </a>
</div>
```
instead of
```tsx
<a>
  <div>
   Button
  </div>
</a>
```